src/toolbars/TimeToolBar.h

```diff
@@ -64,6 +66,20 @@ class TimeToolBar final : public ToolBar
    Observer::Subscription mFormatChangedToFitValueSubscription;
    Observer::Subscription mFormatsSubscription;
 
+   wxMenu *timeToolBarMenu;
+   wxMenuItem *sendMTCCheckBox;
+   void CreateDropdownMenu();
+   void OnCheckboxItemSelected(wxCommandEvent &event);
+   void OnRightClick(wxMouseEvent &event);
+
+   PortMidiStream *portMidiStream;
+   PmError portMidiError;
+   bool isPortMidiInitialized = false;
+   void InitializePortMidi();
+   void TerminatePortMidi();
+   void UpdatePortMidiAccordingToPreferences();
```

Comment:
This function isn’t reached when the state of the checkbox is changed in PlaybackPrefs.

You could call it from the Commit method of the preference page.

Again, move the functions that do all the real business to a file whose job is not GUI.  Neither the toolbar nor preference page should depend on the other.